### PR TITLE
Master tests

### PR DESCRIPTION
**Pull request by jennyfolkesson**
I've updated tests for master. I've also updated the micro_dl codebase since the current master was broken for certain use cases like segmentation and 3D inference.

I've also updated some packages to avoid errors and upgrade soon to be obsolete functions. Updated packages are:
* matplotlib
* pandas
* skimage

While I hope my changes to the codebase will not break anything for the way you're currently running microDL, the tests don't cover everything so it would be great if someone could test and see if everything still runs as expected for some current data. I look forward to your feedback. Thanks!

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,14 +1,14 @@
 Cython==0.29.10
 keras==2.1.6
-matplotlib==3.1.1
+matplotlib==3.5.1
 natsort==6.0.0
 nose==1.3.7
 numpy==1.21.0
 opencv-python==4.2.0.32
-pandas==0.24.2
+pandas==1.1.3
```
Comment (smguo) [inline requirements.txt:8]: We had some issues with pandas 1.1.3. It has a bug that caused issues with reading the metadata CSV file:
https://github.com/pandas-dev/pandas/issues/37094

pandas 1.1.5 seems to work fine

**Review by smguo — APPROVED** [review]
Thanks for fixing the tests and 3D inference @jennyfolkesson! I left some comments. The PR looks good. It might conflict with PR #147 though. Feel free to merge the PR after you merge the changes from #147.

File: micro_dl/inference/image_inference.py [hunk 2]
```diff
@@ -501,15 +523,19 @@ def save_pred_image(self,
...
         """
-        # Write prediction image
-        if self.name_format == 'cztp':
+        if pred_chan_name is None:
+            if 'channel_name' in meta_row:
+                pred_chan_name = meta_row['channel_name']
+        if pred_chan_name != pred_chan_name:
+            # Then it's nan and we don't want that
+            pred_chan_name = None
+        if pred_chan_name is None:
```
[the remark this answers is not in the document]
Comment (jennyfolkesson) [reply micro_dl/inference/image_inference.py:538]: Thanks for catching this. It was an issue where to_csv then read_csv in pandas converts None's to NaNs.
I've now updated it to convert NaNs to None when reading the csv but please let me know if you have a more elegant solution.

[the remark this answers is not in the document]
Comment (smguo) [reply micro_dl/inference/image_inference.py:536]: I understand the issue now. Yes replacing NaN to None when reading the csv would be cleaner

**Comment by jennyfolkesson** [comment]
@smguo @JohannaRahm I rebased this branch with respect to master. I had to make some changes to Johanna's code to make it work with 3D data and metrics orientations other than xy. Please let me know if you'd like to review again, otherwise I'll merge to master.

**Comment by jennyfolkesson** [comment]
Since nobody is jumping on this exciting opportunity to do a re-review, I'm merging. :-)

**Pull request merged**